tests/test_builders/test_build_epub.py

```diff
@@ -34,35 +41,41 @@ class EPUBElementTree:
         'epub': 'http://www.idpf.org/2007/ops',
     }
 
-    def __init__(self, tree):
+    def __init__(self, tree: ET.Element) -> None:
         self.tree = tree
 
     @classmethod
-    def fromstring(cls, string):
+    def fromstring(cls, string: str | bytes) -> Self:
         tree = ET.fromstring(string)  # NoQA: S314  # using known data in tests
         return cls(tree)
 
-    def find(self, match):
+    def find(self, match: str) -> Self:
         ret = self.tree.find(match, namespaces=self.namespaces)
-        if ret is not None:
-            return self.__class__(ret)
-        else:
-            return ret
+        assert ret is not None
+        return self.__class__(ret)
 
-    def findall(self, match):
+    def findall(self, match: str) -> list[Self]:
         ret = self.tree.findall(match, namespaces=self.namespaces)
         return [self.__class__(e) for e in ret]
 
-    def __getattr__(self, name):
-        return getattr(self.tree, name)
+    @property
+    def text(self) -> str | None:
+        return self.tree.text
+
+    @property
+    def attrib(self) -> dict[str, str]:
+        return self.tree.attrib
+
+    def get(self, key: str) -> str | None:
+        return self.tree.get(key)
 
-    def __iter__(self):
+    def __iter__(self) -> Generator[Self, None, None]:
```

Comment:
while this is correct, it might be more clear to return an Iterator rather than a Generator

Reply:
Generator is a larger interface though, we don't support `.send()` here etc.

Reply:
But do we need that interface here? Especially since in this case it would be `send(None)`

Reply:
Using `Iterator` is the right choice, the above was an argument in support

Reply:
Using `Iterator` is the right choice, the above was an argument in support